lib/rpm_package.py

```diff
@@ -255,6 +259,35 @@ def _load(self):
         except TypeError:
             raise exception.PackageDescriptorError(package=self.name)
 
+    def get_spec_macros(self):
+        """
+        Get command line string to define spec file macros externally.
+
+        Returns:
+            str: string to be appended to the rpmbuild command
+        """
+        macros_string = ""
+        for package_attribute, macro_name in (
+                PACKAGE_METADATA_TO_RPM_MACRO_MAPPING.items()):
+            subnode = self.package_data
+            for subnode_id in package_attribute:
+                try:
+                    subnode = subnode[subnode_id]
+                except (KeyError, IndexError):
+                    LOG.debug(
```

Comment:
Some packages do not have commit IDs in their spec files because they correspond exactly to the version's branch in their repository (e.g. golang, systemtap, libnl3).
We also have the `open-power-host-os` package that do not have external sources and, consequently, would fail with no mappings.
I think it's enough to print a message in debug level if no mappings are found.
I am also logging a message when the mapping is found instead of in this `except` block, to avoid too many messages. The mapping not found can be inferred.